server/src/main/scala/io/delta/kernelsharedtable/DeltaSharedTableKernel.scala

```diff
@@ -308,6 +364,145 @@ class DeltaSharedTableKernel(
     throw new DeltaSharingUnsupportedOperationException("not implemented yet")
   }
 
+  private def processBatchByColumnVector(
+      scanFileBatch: FilteredColumnarBatch,
+      version: Long,
+      queryType: QueryTypes.QueryType,
+      respondedFormat: String,
+      table: Table,
+      engine: Engine,
+      isVersionQuery: Boolean,
+      snapshot: SharedTableSnapshot): Seq[Object] = {
+
+    val batchData = scanFileBatch.getData
+    val batchSize = batchData.getSize
+    val addFileVectors = getAddFileColumnVectors(batchData)
+    val selectionVector = scanFileBatch.getSelectionVector
+    var addFileObjects = Seq[Object]()
+    for (rowId <- 0 until batchSize) {
+      val isSelected = !selectionVector.isPresent ||
+        (!selectionVector.get.isNullAt(rowId) && selectionVector.get.getBoolean(rowId))
+      if (isSelected) {
+        val partitionValues = getDeltaPartitionValues(addFileVectors.partitionValues, rowId)
+        val addFileVectorPath = addFileVectors.path.getString(rowId)
+
+        val dataPath = new Path(table.getPath(engine))
```

Comment:
you only need to get this once, consider put it outside the for loop.

Reply:
done